### PR TITLE
update documentation for build-rootfs and remove old files.

### LINKED ISSUE #28140: Update cross build-rootfs for modern world
As part of building docker image for arm64 cross compiling I spent a lot of time looking at build--rootfs.sh.  The following seem like changes which should be done...

+ Add latest Ubuntu version: Artful and Bionic
+ Remove unsupported Ubuntu versions: Vivid and Wily
+ Add latest lldbdev versions: 4.0, 5.0, 6.0
+ Update lldb default to 3.9 to be consistent with non-cross build
+ Drop old lldb versions 3.6 and 3.8
+ Add error checking which is present in core-setup version

See https://github.com/dotnet/core-setup/issues/3843

@janvorli FYI

### PR DESCRIPTION
**Pull request by wfurt**
This is follow-up to Arcade updates (https://github.com/dotnet/arcade/pull/2680 and https://github.com/dotnet/arcade/pull/2772)
I updated build documentation to use eng/common/cross/* and removed cross/*copy  from corefx. 

fixes #28140

### CODE REVIEW

**Comment by janvorli** [comment]
You still need to update the https://github.com/dotnet/corefx/blob/master/src/Native/Unix/gen-buildsys-clang.sh#L54 and similar place in the gcc related one so that they use the files from the new folder.
Could you please also update the title of this change since updating the doc is only part of it?

File: cross/x86_ci_script.sh [hunk 1]
```diff
@@ -1,64 +0,0 @@
-#!/bin/bash
```
Comment (janvorli) [inline cross/x86_ci_script.sh]: Was the deletion of this script intentional? It doesn't seem to exist in the eng/common/cross folder.

Comment (wfurt) [reply cross/x86_ci_script.sh]: yes, I realized that. I assumed all needed was moved over but it was not. I'm trying to find out how to undelete it. I may abandon this PR and create new and  remove only files related to build-rootfs

**Comment by wfurt** [comment]
any more thoughts on this @janvorli ? 
We may do more cleanup to move more to Arcade but this at least finishes docs and  build-rootfs part.

**Comment by janvorli** [comment]
@wfurt, it seems good to me. I assume you have tested that when you build a rootfs without explicitly specifying the rootfs dir, the cross build uses that without the need to set the ROOTFS_DIR.

**Comment by stephentoub** [comment]
@wfurt, can/should this be merged?

**Comment by wfurt** [comment]
I'll try to get it in today @stephentoub. I wanted to double-check the last question. I think i did that test but I wanted to be sure.

**Comment by wfurt** [comment]
build-rootfs.sh in Arcade is executable but clone in corefx is not. I don't know if that is some issue with maestro merges. cc: @wtgodbe @ericstj @safern 

Aside from that when running without parameter works but it will default to eng/common/cross instead of cross @janvorli 
So I will either need to fix it in the scrip or update corefx to look in new location. I don't know what is better at this point.

**Comment by ericstj** [comment]
> build-rootfs.sh in Arcade is executable but clone in corefx is not. I don't know if that is some issue with maestro merges

Similar to https://github.com/dotnet/arcade/issues/1535.  @jcagme @mmitche should darc be flowing permissions?

**Comment by mmitche** [comment]
@ericstj Yes, it does, though there may be some corner cases in there that might be problematic. We made some fixes in this area.

The primary corner case which might be a problem would be if executable permissions we fixed in arcade prior to us fixing the executable bit flow issues.

Either way, you should be able to fix this manually without issue.

**Comment by ViktorHofer** [comment]
> build-rootfs.sh in Arcade is executable but clone in corefx is not. I don't know if that is some issue with maestro merges.

If this still happens, as a workaround you should be able to change the permissions manually of the files in eng/common/cross in corefx and git add them.

**Comment by ViktorHofer** [comment]
> So I will either need to fix it in the scrip or update corefx to look in new location. I don't know what is better at this point.

Sounds like you want an arcade extension point in eng/common/cross. If you define one it would be great if remaining cross folder would go into eng/cross.

**Comment by maryamariyan** [comment]
@wfurt this PR hasn't been active for a while. Would you be updating it in the next couple of days? or would you like to close and reopen at a later time?

**Comment by ViktorHofer** [comment]
@wfurt is it possible to move the cross folder under eng?

**Comment by wfurt** [comment]
yes, I think that should be the goal. I have some more work in Arcade but it was on hold because of 3.0 and my vacation.

**Comment by ViktorHofer** [comment]
OK got it. Is there already an extension point defined in arcade? If yes, is that currently `$(RepoRoot)\cross`? It would be great to create/change it to `$(RepoRoot)\eng\cross`.

**Pull request merged**